src/loader/AutoLoginImGui.cpp

```diff
@@ -0,0 +1,2010 @@
+			if (Character)
+			{
+				if (ImGui::Button("Hotkey"))
+				{
+					hot_key = hotkey;
+					LauncherImGui::OpenModal("Input Hotkey");
+				}
+
+				ShowHotkeyWindow("Input Hotkey", hot_key,
+					[this] { hotkey = hot_key; });
+
+				ImGui::SameLine();
+				ImGui::Text("%s", !hotkey.empty() ? hotkey.c_str() : "<None>");
+
+				if (ImGui::Button("EQ Path"))
+				{
+					eq_path = eqPath;
+					LauncherImGui::OpenModal("Input EQ Path");
+				}
+
+				SetEQDirModal(eq_path, [this] { eqPath = eq_path; });
+
+				ImGui::SameLine();
+				ImGui::Text("%s", eqPath ? eqPath->c_str() : "<Default>");
+
+				DefaultOptional<int>(charSelectDelay, "Override Character Select Delay",
+					[]()
+					{
+						if (const auto delay = login::db::ReadSetting("char_select_delay"))
+							return GetIntFromString(*delay, 3);
+						return 3;
+					},
+					[](int& char_select_delay)
+					{
+						return ImGui::InputScalar("##Character Select Delay", ImGuiDataType_U32, &char_select_delay);
+					});
+
+				DefaultOptional<bool>(endAfterSelect, "Override End After Character Select",
+					[]()
+					{
+						if (const auto end = login::db::ReadSetting("end_after_select"))
+							return GetBoolFromString(*end, false);
+						return false;
+					},
+					[](bool& end_after_select)
+					{
+						bool changed = false;
+						if (ImGui::BeginCombo("##End After Character Select", end_after_select ? "enabled" : "disabled"))
+						{
+							if (ImGui::Selectable("enabled", end_after_select))
+							{
+								end_after_select = true;
+								changed = true;
+							}
+
+							if (ImGui::Selectable("disabled", !end_after_select))
+							{
+								end_after_select = false;
+								changed = true;
+							}
+
+							ImGui::EndCombo();
+						}
+
+						return changed;
+					});
+
+				if (ImGui::Button("Custom Character INI"))
+				{
+					custom_ini = customClientIni;
+					LauncherImGui::OpenModal("Input Character INI File");
+				}
+
+				SetEQFileModal("Input Character INI File", custom_ini, "eqclient.ini",
+					[this] { customClientIni = custom_ini; });
+
+				ImGui::SameLine();
+				ImGui::Text("%s", customClientIni ? customClientIni->string().c_str() : "<Default>");
+			}
+		}
+
+		DefaultModalButtons(ok_action);
+
+		LauncherImGui::EndModal();
+	}
+}
+
+void ProfileTable(const ProfileGroupInfo& info)
+{
+	static ProfileInfo selected;
+	static std::string remove_message;
+
+	if (ImGui::BeginTable("Main List", 7, ImGuiTableFlags_SizingStretchProp | ImGuiTableFlags_Borders | ImGuiTableFlags_NoBordersInBody))
+	{
+		if (!info.profileName.empty())
+		{
+			ImGui::TableSetupColumn("Account");
+			ImGui::TableSetupColumn("Server Type");
+			ImGui::TableSetupColumn("Server");
+			ImGui::TableSetupColumn("Character");
+			ImGui::TableSetupColumn("Persona");
+			ImGui::TableSetupColumn("Hotkey");
+			ImGui::TableSetupColumn("##play");
+			ImGui::TableHeadersRow();
+
+			for (auto& profile : login::db::GetProfiles(info.profileName))
+			{
+				ImGui::PushID(&profile);
```

Comment:
This PushID isn't working because profile is a value type and is always occupying the same space on the stack. The reference and auto is misleading.

Best solve I have is to just count rows with an index.

Reply:
The reference actually makes this const reference. Which makes profile immutable, which I eblieve you intended for Selectable down below to mutate the checked flag. (otherwise the selectable doesn't make sense -- how do you toggle the checked state?)